specification/ai/Face/models.common.tsp

```diff
@@ -0,0 +1,218 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+
+using Azure.Core;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+
+namespace Face;
+
+@doc("The ApiVersion path parameter.")
+model ApiVersionPathParameter {
+  @segment("face")
+  @path("apiVersion")
+  @doc("The API version to use for this operation.")
+  apiVersion: string;
+}
+
+@doc("The request for list resources.")
+model ListRequestOptions {
+  @doc("List resources greater than the \"start\". It contains no more than 64 characters. Default is empty.")
+  @query
+  start?: string;
+
+  @doc("The number of items to list, ranging in [1, 1000]. Default is 1000.")
+  @query
+  @minValue(1)
+  @maxValue(1000)
+  top?: int32;
+}
+
+@doc("The recognition model for the face.")
+enum RecognitionModel {
+  @doc("The default recognition model for Face - Detect. All those faceIds created before 2019 March are bonded with this recognition model.")
+  recognition_01,
+
+  @doc("Recognition model released in 2019 March.")
+  recognition_02,
+
+  @doc("Recognition model released in 2020 May.")
+  recognition_03,
+
+  @doc("Recognition model released in 2021 February. It's recommended to use this recognition model for better recognition accuracy.")
+  recognition_04,
+}
+
+@doc("The detection model for the face.")
+enum DetectionModel {
+  @doc("The default detection model for Face - Detect. Recommend for near frontal face detection. For scenarios with exceptionally large angle (head-pose) faces, occluded faces or wrong image orientation, the faces in such cases may not be detected.")
+  detection_01,
+
+  @doc("Detection model released in 2019 May with improved accuracy especially on small, side and blurry faces.")
+  detection_02,
+
+  @doc("Detection model released in 2021 February with improved accuracy especially on small faces.")
+  detection_03,
+}
+
+@doc("Model for object name and user data.")
+model NameAndUserData {
+  @doc("User defined name, maximum length is 128.")
+  name: string;
+
+  @doc("Optional user defined data. Length should not exceed 16KB.")
+  userData?: string;
+}
+
+@doc("Common model for persisted face.")
+@resource("persistedfaces")
+model BaseFace {
+  @key
+  @visibility("read")
+  @doc("Face ID of the face.")
+  persistedFaceId: string;
+
+  @doc("User-provided data attached to the face. The size limit is 1KB.")
+  userData?: string;
+}
+
+@doc("Query parameter for adding face to person.")
+model AddFaceOptions {
+  @query
+  @doc("A face rectangle to specify the target face to be added to a person, in the format of 'targetFace=left,top,width,height'.")
+  targetFace?: string;
+
+  @query
+  @doc("The 'detectionModel' associated with the detected faceIds. Supported 'detectionModel' values include 'detection_01', 'detection_02' and 'detection_03'. The default value is 'detection_01'.")
+  detectionModel?: DetectionModel;
```

Comment:
server side default could be written as `detectionModel?: DetectionModel = DetectionModel.detection_01;`

Reply:
Fixed, also applied to most of request parameter.